packages/earn-controller/src/EarnController.ts

```
@@ -0,0 +1,334 @@
+      name: controllerName,
+      metadata: earnControllerMetadata,
+      messenger,
+      state: {
+        ...getDefaultEarnControllerState(),
+        ...state,
+      },
+    });
+
+    // Initialize with current network
+    this.#initializeSDK();
+
+    // Listen for network changes
+    this.messagingSystem.subscribe(
+      'NetworkController:networkDidChange',
+      ({ selectedNetworkClientId }) => {
+        this.#initializeSDK(selectedNetworkClientId);
+        this.refreshPooledStakingData().catch(console.error);
+      },
+    );
+
+    // Listen for account changes
+    this.messagingSystem.subscribe(
+      'AccountsController:selectedAccountChange',
+      () => {
+        this.refreshPooledStakingData().catch(console.error);
+      },
+    );
+  }
+
+  #initializeSDK(networkClientId?: string) {
+    const { selectedNetworkClientId } = networkClientId
+      ? { selectedNetworkClientId: networkClientId }
+      : this.messagingSystem.call('NetworkController:getState');
+
+    const networkClient = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      selectedNetworkClientId,
+    );
+
+    if (!networkClient?.provider) {
+      this.#stakeSDK = null;
+      return;
+    }
+
+    const provider = new Web3Provider(networkClient.provider);
+    const { chainId } = networkClient.configuration;
+
+    // Initialize appropriate contracts based on chainId
+    const config: StakeSdkConfig = {
+      chainId: convertHexToDecimal(chainId),
+    };
+
+    try {
+      this.#stakeSDK = StakeSdk.create(config);
+      this.#stakeSDK.pooledStakingContract.connectSignerOrProvider(provider);
+    } catch (error) {
+      this.#stakeSDK = null;
+      // Only log unexpected errors, not unsupported chain errors
+      if (
+        !(
+          error instanceof Error &&
+          error.message.includes('Unsupported chainId')
+        )
+      ) {
+        console.error('Stake SDK initialization failed:', error);
+      }
```

It looks like this is suppressing all errors even if some are logged. Is that intentional? Should there be this at least?

``` suggestion
      } else {
        throw error;
      }

---

I intentionally supressed the errors during the initialization of the constructor as I am not quite sure if we want to bubble this all the way up to the app's error boundary. 
I am trying to find how other controllers are handling construction error but don't see a lot of handling this. @mcmire Is there a suggested pattern around when handling errors?